pkg/endpointmanager/manager.go

```diff
@@ -432,6 +432,16 @@ func (mgr *EndpointManager) RegenerateAllEndpoints(regenMetadata *regeneration.E
 	return &wg
 }
 
+// OverrideEndpointOpts applies the given options to all endpoints.
+func (mgr *EndpointManager) OverrideEndpointOpts(om option.OptionMap) error {
+	for _, ep := range mgr.GetEndpoints() {
+		if _, err := ep.ApplyOpts(om); err != nil {
+			return err
```

Comment:
^ May need something like this so you have an error to compare correctly:

```
diff --git a/pkg/endpoint/lock.go b/pkg/endpoint/lock.go
index f9f35a1b140a..51c7f0747920 100644
--- a/pkg/endpoint/lock.go
+++ b/pkg/endpoint/lock.go
@@ -16,12 +16,16 @@ package endpoint

 import "fmt"

+var (
+       ErrEndpointDeleted = fmt.Errorf("lock failed: endpoint is in the process of being removed")
+)
+
 // lockAlive returns error if endpoint was removed, locks underlying mutex otherwise
 func (e *Endpoint) lockAlive() error {
        e.mutex.Lock()
        if e.IsDisconnecting() {
                e.mutex.Unlock()
-               return fmt.Errorf("lock failed: endpoint is in the process of being removed")
+               return ErrEndpointDeleted
        }
        return nil
 }
```